assets/behaviors/creatures/combinedCritter.behavior

```diff
@@ -0,0 +1,107 @@
+{
+selector: [
+  {
+    guard: {
+
+      // Territory would to be implemented as Component
+      // Implementation of territorial Critter
+      componentPresent: "Behaviours:Territory",
+      values: ["F Territory != null"],
+      "child": {
+        "sequence": [
+          {
+            "lookup": {
+              "tree": "territorialCritter.behavior"
+            }
+          }
+        ]
+      }
+    }
+
+  // First Guard fo Selector for Territory End.
+    guard: { // Implementation of scared Critter
+        componentPresent: "Behaviors:FindNearbyPlayers",
+        values: ["N charactersWithinRange != null"], // Adjust for Nemesis / Scary Creatures
```

Comment:
We discussed this in the meeting today.

The `N` is a parsing directive that specifiies whether the condition has the form `type fieldName operator` or `type fieldName operator valueOrReference`.

Technically, we could use `V charactersWithinRange != null` to ensure that the list itself is defined and not `null` (checking agains the _value_ `null`). However, this is not the semantics we are looking for.

Therefore, using the `nonEmpty` operator on the collection with no value to check against (`N`) is the solution:
```suggestion
        values: ["N charactersWithinRange nonEmpty"], // Adjust for Nemesis / Scary Creatures
```